libr/util/idpool.c

```diff
@@ -53,18 +72,34 @@ R_API bool r_id_pool_kick_id(RIDPool* pool, ut32 kick) {
 	}
 	if (kick == (pool->next_id - 1)) {
 		pool->next_id--;
+		if (pool->freed_ids) {
+			RBTreeIter iter = r_rbtree_last (pool->freed_ids);
+			while ((iter.len) && ((pool->next_id - 1) <= (ut32)(size_t)(iter.path[iter.len-1]->data))) {
```

Comment:
I should create an API to make fetching the data associated with the current position easier `iter.path[iter.len-1]->data`

I have an idea to make rbtree intrusive ...

struct YourStruct { struct r_rbtree_t rb; ... }